webapp/handlers.py

```diff
@@ -52,6 +52,14 @@ def page_not_found(error):
 
         return flask.render_template("404.html", error=error.description), 404
 
+    @app.errorhandler(500)
+    @app.errorhandler(501)
+    @app.errorhandler(502)
+    @app.errorhandler(504)
+    @app.errorhandler(505)
+    def internal_error(error):
+        return flask.render_template("50X.html", error=error), error.code
```

Comment:
You're expecting an `HTTPError` width `name` and `code` properties, but it's really quite likely to be any random Python error, like a `ValueError` or `DatabaseError`, which won't have these properties.

I suggest you use a function like this, which caters for any kind of error:

``` python3
    @app.errorhandler(500)
    @app.errorhandler(501)
    @app.errorhandler(502)
    @app.errorhandler(504)
    @app.errorhandler(505)
    def internal_error(error):
        error_name = getattr(error, "name", type(error).__name__)

        return_code = getattr(error, "code", 500)

        return (
            flask.render_template("50X.html", error_name=error_name),
            return_code,
        )
```